### PR TITLE
Respond rejection on start instance to client on failure to subscribe to events

### LINKED ISSUE #10250: Rejection on start instance is not always responded to client
**Describe the bug**

The rejection is only written on the log and not responded to in case a CreateProcessInstance with start instructions is rejected due to 
- #9422

See [CreateProcessInstanceProcessor.java#L120-L123](https://github.com/camunda/zeebe/blob/main/engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceProcessor.java#L120-L123)

**Expected behavior**

The rejection is also responded to the client.

### PR DESCRIPTION
**Pull request by korthout**
## Description

The rejection is now also responded to the client, in case a CreateProcessInstance with start instructions is rejected due to
- https://github.com/camunda/zeebe/issues/9422

This PR additionally adds a QA test to verify the integration between the java client and the engine for creating process instances with start instructions.

## Related issues

closes #10250 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/CreateProcessInstanceTest.java [hunk 1]
```diff
@@ -46,6 +47,17 @@ public void deployProcess() {
...
         CLIENT_RULE.deployProcess(Bpmn.createExecutableProcess(processId).startEvent("v1").done());
     secondProcessDefinitionKey =
+        CLIENT_RULE.deployProcess(
+            Bpmn.createExecutableProcess(processId)
+                .eventSubProcess(
+                    "event-sub",
+                    e ->
+                        e.startEvent("msg-start-event")
+                            .message(msg -> msg.name("msg").zeebeCorrelationKey("=missing_var")))
+                .startEvent("v3")
```
Comment (remcowesterhoud) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/CreateProcessInstanceTest.java:57]: 🔧 I think it's confusing to have the 2nd process definition be V3, and the 3rd process definition be V2. I'm not sure I understand the reasoning you mentioned in the commit. If it's a new deployment, how would it interfere with other tests?

**Review by remcowesterhoud — APPROVED** [review]
LGTM @korthout 👍 

I do have one comment / question. Maybe I just don't fully understand the QA tests.

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/CreateProcessInstanceTest.java (hunk 1 above)
Comment (korthout) [reply qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/CreateProcessInstanceTest.java:57]: 👍 I agree. I guess I just wanted to continue on what's there, but it's confusing. I'll just introduce a new process altogether.

**Comment by korthout** [comment]
@remcowesterhoud I've reverted the other tests by deploying a completely different process. Please have another look

**Review by remcowesterhoud — APPROVED** [review]
Thanks for the changes @korthout 👍 

bors merge

**Pull request merged**